crates/rpc/src/server/accept.rs

```diff
@@ -162,61 +193,88 @@ impl<'a> TryFrom<&'a str> for AcceptHeaderValue<'a> {
+    #[case::with_matching_network(&[
+        ("miden-network", TEST_GENESIS_COMMITMENT)
+    ])]
+    #[case::with_matching_network_and_version(&[
+        ("miden-network", TEST_GENESIS_COMMITMENT),
+        ("accept", "application/vnd.miden.1.2.3+grpc")
+    ])]
     #[test]
-    fn greater_minor_does_not_match() {
-        let a = AcceptLayer::new().unwrap();
-        let version = Version::parse("0.99.0").unwrap();
-        assert!(!a.version_req.matches(&version));
-    }
+    fn request_should_pass(#[case] headers: &[(&'static str, &'static str)]) {
+        let headers = headers
+            .iter()
+            .map(|(k, v)| (HeaderName::from_static(k), HeaderValue::from_static(v)));
+        let headers = HeaderMap::from_iter(headers);
 
-    #[test]
-    fn lower_minor_does_not_match() {
-        let a = AcceptLayer::new().unwrap();
-        let version = Version::parse("0.1.0").unwrap();
-        assert!(!a.version_req.matches(&version));
+        let uut = HeaderVerificationLayer::for_tests();
+        uut.verify(&headers).unwrap();
     }
 
+    #[rstest::rstest]
+    #[case::with_older_minor_version(&[
+        ("accept", "application/vnd.miden.1.1.100+grpc")
+    ])]
+    #[case::with_newer_minor_version(&[
+        ("accept", "application/vnd.miden.1.3.0+grpc")
+    ])]
+    #[case::with_older_minor_version_and_matching_network(&[
+        ("accept", "application/vnd.miden.1.1.100+grpc"),
+        ("miden-network", TEST_GENESIS_COMMITMENT),
+    ])]
+    #[case::with_network_mismatch(&[
+        ("miden-network", "0x00000000000000000000000000000000000000000000000000000000dead1234"),
+    ])]
+    #[case::with_network_mismatch_but_matching_version(&[
+        ("accept", "application/vnd.miden.1.2.3+grpc"),
+        ("miden-network", "0x00000000000000000000000000000000000000000000000000000000dead1234"),
+    ])]
+    #[case::invalid_version(&[
+        ("accept", "application/vnd.miden.x.y.z+grpc"),
+    ])]
+    #[case::without_grpc_version_suffix(&[
+        ("accept", "application/vnd.miden.1.2.3"),
+    ])]
     #[test]
-    fn greater_major_does_not_match() {
-        let a = AcceptLayer::new().unwrap();
-        let version = Version::parse("9.9.0").unwrap();
-        assert!(!a.version_req.matches(&version));
-    }
+    fn request_should_be_rejected(#[case] headers: &[(&'static str, &'static str)]) {
+        let headers = headers
+            .iter()
+            .map(|(k, v)| (HeaderName::from_static(k), HeaderValue::from_static(v)));
+        let headers = HeaderMap::from_iter(headers);
 
-    #[test]
-    fn valid_accept_header_is_parsed() {
-        let input = "application/vnd.miden.1.0.0+grpc";
-        let expected = AcceptHeaderValue {
-            app_name: "miden",
-            version: "1.0.0",
-            response_type: "grpc",
-        };
-        assert_eq!(AcceptHeaderValue::try_from(input), Ok(expected));
+        let uut = HeaderVerificationLayer::for_tests();
+        uut.verify(&headers).unwrap_err();
```

Comment:
Could we test the exact error variants coming out? Given you have created the error enums now.

Reply:
> For example, in the case where both mismatch, which error should it be? Could be either..

It would be well defined which error would be returned if both mismatched. Checking the error proves our logic is working precisely as expected. We could erroneously return invalid semver variant for an input with a correct semver for example. Unless I'm missing something.

Reply:
I had a crack at adding the expected error variants into the cases but I think its blocked on variants like 
```rust
    VersionParsing(#[source] semver::Error, String),
```

Because `semver::Error` and its constructors etc are not public so we can't recreate them for the tests.